rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -1969,22 +1987,29 @@ bool spu_thread::process_mfc_cmd()
 					{
 						*reinterpret_cast<atomic_t<u32>*>(&data) += 0;
 
-						auto& super_data = *vm::get_super_ptr<decltype(rdata)>(addr);
-
-						// Full lock (heavyweight)
-						// TODO: vm::check_addr
-						vm::writer_lock lock(addr);
+						const auto render = rsx::get_current_renderer();
+						const bool lock_rsx = render->iomap_table.io[addr >> 20] != -1 && g_cfg.core.rsx_accurate_res_access;
```

Comment:
Consider reordering this condition if checking iomap_table is more expensive than checking the cfg variable (indexed ld+shl+cmp vs cmp) since the config var can be assumed to virtually always be false.